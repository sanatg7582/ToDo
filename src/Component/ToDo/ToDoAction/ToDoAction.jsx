import React from "react";
import PropTypes from "prop-types";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faTrash, faPen } from "@fortawesome/free-solid-svg-icons";
import Button from "../../Shared/Button";
/**
 * This function return Edit and Delete button
 * @param {number} index
 * @param {func} handleOnClickRemove
 * @param {func} handleOnClickUpdate
 * @returns Button Element
 */
const ToDoAction = props => {
  const { index, handleOnClickRemove, handleOnClickUpdate } = props;
  return (
    <>
      <Button
        buttonType="button"
        btnClass="warning"
        handleOnClick={handleOnClickUpdate}
        index={index}
      >
        <FontAwesomeIcon icon={faPen} />
      </Button>
      &nbsp;
      <Button
        buttonType="button"
        btnClass="danger"
        handleOnClick={handleOnClickRemove}
        index={index}
      >
        <FontAwesomeIcon icon={faTrash} />
      </Button>
    </>
  );
};
// default props value
ToDoAction.defaultProps = {
  index: 0,
  handleOnClickRemove: () => {},
  handleOnClickUpdate: () => {}
};
//prop type validation
ToDoAction.propTypes = {
  index: PropTypes.number,
  handleOnClickRemove: PropTypes.func.isRequired,
  handleOnClickUpdate: PropTypes.func.isRequired
};
export default ToDoAction;
