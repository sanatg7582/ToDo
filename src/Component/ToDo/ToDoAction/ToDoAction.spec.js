import React from "react";
import { mount } from "enzyme";
import ToDoAction from "./ToDoAction";

describe("Load <ToDoAction />", () => {
  let wrapper = "";
  const handleOnClickUpdate = jest.fn();
  const handleOnClickRemove = jest.fn();
  beforeAll(() => {
    const props = {
      handleOnClickUpdate: handleOnClickUpdate,
      handleOnClickRemove: handleOnClickRemove,
      index: 0
    };
    wrapper = mount(<ToDoAction {...props} />);
  });

  it("renders without crashing", () => {
    expect(wrapper).toBeTruthy();
  });

  it("checking button length", () => {
    expect(wrapper.find("button")).toHaveLength(2);
  });

  it("checking first button props", () => {
    expect(
      wrapper
        .find("button")
        .first()
        .type()
    ).toEqual("button");
    expect(
      wrapper
        .find("button")
        .first()
        .hasClass("btn warning")
    ).toEqual(true);
  });
  it("Checking last button props", () => {
    expect(
      wrapper
        .find("button")
        .last()
        .type()
    ).toEqual("button");
    expect(
      wrapper
        .find("button")
        .last()
        .hasClass("btn danger")
    ).toEqual(true);
  });
  it("checking first button click", () => {
    wrapper
      .find("button")
      .first()
      .simulate("click");
    expect(handleOnClickUpdate).toHaveBeenCalled();
  });
  it("checking last button click", () => {
    wrapper
      .find("button")
      .last()
      .simulate("click");
    expect(handleOnClickRemove).toHaveBeenCalled();
  });
  it("SnapShot test", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
