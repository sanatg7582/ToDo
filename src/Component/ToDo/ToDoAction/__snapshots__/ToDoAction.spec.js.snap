// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Load <ToDoAction /> SnapShot test 1`] = `ReactWrapper {}`;
